conda/core/subdir_data.py

```diff
@@ -262,31 +292,27 @@ def _iter_records_by_name(self, name):
         for i in self._names_index[name]:
             yield self._package_records[i]
 
-    def _load_state(self) -> dict:
+    def _load_state(self):
         """
         Cache headers and additional data needed to keep track of the cache are
         stored separately, instead of the previous "added to repodata.json"
         arrangement.
         """
-        try:
-            state_path = pathlib.Path(self.cache_path_state)
-            log.debug("Load %s cache from %s", self.repodata_fn, state_path)
-            # efficient according to scalene profiler; about equal to
-            # json.loads(state_path.read_text()) and better that open("rb")
-            with state_path.open("r") as s:
-                state = json.load(s)
-            return state
-        except (json.JSONDecodeError, OSError):
-            log.debug("Could not load state", exc_info=True)
-            return {}
+        return self.repo_cache.load_state()
 
-    def _save_state(self, state: dict):
-        return pathlib.Path(self.cache_path_state).write_text(json.dumps(state, indent=True))
+    def _save_state(self, state: RepodataState):
+        assert Path(state.cache_path_json) == Path(self.cache_path_json)
+        assert Path(state.cache_path_state) == Path(self.cache_path_state)
+        assert state.repodata_fn == self.repodata_fn
+        return state.save()
 
     def _load(self):
-        try:
-            mtime = getmtime(self.cache_path_json)
-        except OSError:
+        cache = self.repo_cache
+        cache.load_state()  # XXX should this succeed even if FileNotFound?
```

Comment:
Pretty sure this does succeed on file not found, with RepoCache/RepodataState creating the state file if it is not available.